packages/internal-test-env/index.ts

```diff
@@ -143,113 +141,147 @@ function getTestingEnvironment(
-      );
-    }
-  }
-
-  const base = {
-    idp: process.env.E2E_TEST_IDP,
-    environment: process.env.E2E_TEST_ENVIRONMENT,
-    clientCredentials: {
-      owner: {
-        id: process.env.E2E_TEST_OWNER_CLIENT_ID,
-        secret: process.env.E2E_TEST_OWNER_CLIENT_SECRET,
-      },
-    },
-    features: featuredFlags,
-  };
+  return getBaseTestingEnvironment({
+    ...libVars,
+    // Enforce client credentials are present for the resource owner.
+    clientCredentials: { owner: { id: true, secret: true }}
+  });
+}
 
-  return libVars ? merge(base, validateLibVars(libVars)) : base;
+export function getBrowserTestingEnvironment(
+  libVars?: LibraryVariables
+): TestingEnvironmentBrowser {
+  return getBaseTestingEnvironment({
+    ...libVars,
+    // Enforce login/password are present for the resource owner.
+    clientCredentials: { owner: { login: true, password: true }}
+  });
 }
 
 export interface LibraryVariables {
-  notificationGateway?: string;
-  notificationProtocol?: string;
-  vcProvider?: string;
+  notificationGateway?: boolean;
+  notificationProtocol?: boolean;
+  vcProvider?: boolean;
   clientCredentials?: {
     owner?: {
-      id?: string;
-      secret?: string;
-      login?: string;
-      password?: string;
+      id?: boolean;
+      secret?: boolean;
+      login?: boolean;
+      password?: boolean;
     };
     requestor?: {
-      id?: string;
-      secret?: string;
+      id?: boolean;
+      secret?: boolean;
+    };
+  };
+}
+
+export interface BrowserVariables extends LibraryVariables {
+  clientCredentials?: {
+    owner: {
+      login: true;
+      password: true;
+    };
+  };
+}
+
+export interface NodeVariables extends LibraryVariables {
+  clientCredentials?: {
+    owner: {
+      id: true;
+      secret: true;
     };
   };
 }
 
 function validateLibVars(vars: LibraryVariables): object {
+  // Notifications-related environment variables.
   if (
-    typeof vars.notificationGateway !== "undefined" &&
-    typeof vars.notificationGateway !== "string"
+    vars.notificationGateway &&
+    typeof process.env.E2E_TEST_NOTIFICATION_GATEWAY !== "string"
   ) {
     throw new Error(
       "Missing the E2E_TEST_NOTIFICATION_GATEWAY environment variable"
     );
   }
+  if (
+    vars.notificationProtocol &&
+    typeof process.env.E2E_TEST_NOTIFICATION_PROTOCOL !== "string"
+  ) {
+    throw new Error(
+      "Missing the E2E_TEST_NOTIFICATION_PROTOCOL environment variable"
+    );
+  }
+  // VC-related environment variables.
+  if (
+    vars.vcProvider &&
+    typeof process.env.E2E_TEST_VC_PROVIDER !== "string"
+  ) {
+    throw new Error(
+      "Missing the E2E_TEST_NOTIFICATION_PROTOCOL environment variable"
+    );
+  }
+  // Resource owner static credentials.
   if (
     vars.clientCredentials?.owner?.id &&
-    typeof vars.clientCredentials.owner.id !== "string"
+    typeof process.env.E2E_TEST_OWNER_CLIENT_ID !== "string"
```

Comment:
```suggestion
    typeof process.env.E2E_TEST_OWNER_CLIENT_ID !== "string" && process.env.E2E_TEST_OWNER_CLIENT_ID !== ""
```

We should be checking that we actually have a valid value set.

Reply:
Fixed in db46a19900d96aadb5dc6eea9d0f49fad07a0bb1.